include/zephyr/pm/policy.h

```diff
@@ -27,37 +28,29 @@ extern "C" {
  */
 
 /**
- * @brief Callback to notify when maximum latency changes.
- *
- * @param latency New maximum latency. Positive value represents latency in
- * microseconds. SYS_FOREVER_US value lifts the latency constraint. Other values
- * are forbidden.
- */
-typedef void (*pm_policy_latency_changed_cb_t)(int32_t latency);
-
-/**
- * @brief Latency change subscription.
+ * @brief Latency request.
  *
  * @note All fields in this structure are meant for private usage.
  */
-struct pm_policy_latency_subscription {
+struct pm_policy_latency_request {
 	/** @cond INTERNAL_HIDDEN */
 	sys_snode_t node;
-	pm_policy_latency_changed_cb_t cb;
+	uint32_t value_us;
+	struct onoff_client cli;
+	void *internal;
+	uint32_t flags;
```

Comment:
`onoff_client` and two 32-bit variables per latency requestor seem like high memory overhead. Wouldn't it be enough to track these data per subscriber instead of per requestor?

Reply:
if immediate action/subscription is needed then it is required. We can add kconfig option which enables that feature and have those fields conditionally compiled.